cdc/sink/codec/craft/codec_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package craft
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/pingcap/check"
+	"github.com/pingcap/ticdc/pkg/util/testleak"
+)
+
+var _ = check.Suite(&codecSuite{allocator: NewSliceAllocator(64)})
+
+func Test(t *testing.T) { check.TestingT(t) }
+
+type codecSuite struct {
+	allocator *SliceAllocator
+}
+
+func (s *codecSuite) TestSizeTable(c *check.C) {
+	defer testleak.AfterTest(c)()
+	tables := [][]uint64{
+		{
+			1, 3, 5, 7, 9,
+		},
+		{
+			2, 4, 6, 8, 10,
+		},
+	}
+	bits := make([]byte, 16)
+	rand.Read(bits)
+	bits = encodeSizeTables(bits, tables)
+
+	size, decoded, err := decodeSizeTables(bits, s.allocator)
+	c.Check(err, check.IsNil)
+	c.Check(decoded, check.DeepEquals, tables)
+	c.Check(size, check.Equals, len(bits)-16)
+}
+
+func (s *codecSuite) TestUvarintReverse(c *check.C) {
+	defer testleak.AfterTest(c)()
+
+	var i uint64 = 0
+
+	for i < 0x8000000000000000 {
+		bits := make([]byte, 16)
+		rand.Read(bits)
```

Comment:
Could you seed a unix timestamp? Otherwise, it may always read a deterministic sequence.

Reply:
Sure, it's fixed now.